src/problem.ts

```diff
@@ -1,62 +1,78 @@
-import { httpObject, ProblemInterface } from "types";
+import { DEFAULT_KEYS } from "./constants";
+import {
+  httpObject,
+  ProblemInterface,
+  ToJsonParamType,
+  UpdateProblemParamType,
+} from "./types";
 
-enum COPY_MODE {
-    SKIP_PROPS = 'skipProps',
-    LEAVE_PROPS = 'leaveProps'
+export enum COPY_MODE {
+  SKIP_PROPS = "skipProps",
+  LEAVE_PROPS = "leaveProps",
 }
 
 export class Problem extends Error implements ProblemInterface {
-    public type: string;
-    public title: string;
-    public instance?: string;
-    public detail?: string;
-    public http?: httpObject
-    [key: string]: any;
-
-
-    constructor(problem: ProblemInterface, customKeys?: string[]) {
-        super(problem.detail || problem.title);
-        this.http = problem.http
-        this.type = problem.type
-        this.title = problem.title;
-        this.detail = problem.detail;
-        this.instance = problem.instance;
-        this.stack = problem.stack;
-        customKeys?.map((customKey) => {
-            this[customKey] = problem[customKey];
-        })
-    }
+  public type: string;
+  public title: string;
+  public instance?: string;
+  public detail?: string;
+  public http?: httpObject;
+  [key: string]: any;
 
-   copy(problem: ProblemInterface, mode: COPY_MODE, props: string[]): Problem {
-        switch (mode) {
-
-            case COPY_MODE.LEAVE_PROPS:
-                return new Problem(problem, props);
-
-            case COPY_MODE.SKIP_PROPS:
-            default:
-                let keysToBeCopied: string[] = [];
-                for (let key in problem) {
-                    if (props.includes(key))
-                        continue;
-                    keysToBeCopied.push(key)
-                }
-                return new Problem(problem, keysToBeCopied)
-        }
-    };
+  constructor(problem: ProblemInterface, customKeys?: string[]) {
+    super(problem.detail || problem.title);
+    this.http = problem.http;
+    this.type = problem.type;
+    this.title = problem.title;
+    this.detail = problem.detail;
+    this.instance = problem.instance;
+    this.stack = problem.stack;
 
-   toJSON(problem: Problem, includeStack = false): ProblemInterface {
+    // add extra keys
+    Object.keys(problem)
+      .filter((el) => !DEFAULT_KEYS.includes(el))
+      .forEach((k) => (this[k] = problem[k]));
+  }
 
-        const { name, message, stack, ...rest } = problem;
+  copy(problem: ProblemInterface, mode: COPY_MODE, props: string[]): Problem {
```

Comment:
Why we need "ProblemInterface" if that function is prototype method, not static? We should attach some default value to the `mode` argument and make `props` optional. Also, for `COPY_MODE.LEAVE_PROPS` we should "leave" props - we only pass props with problem to the constructor (second argument should be removed) - and also fix `COPY_MODE.SKIP_PROPS` :)

Sorry, I didn't realise such a bugs in previous review!

Arguments should looks like:

```suggestion
  copy(mode: COPY_MODE = COPY_MODE.LEAVE_PROPS, props: string[] = []): Problem {
```